tests/compat.sh

```diff
@@ -1001,10 +989,21 @@ SKIP_NEXT="NO"
 
 trap cleanup INT TERM HUP
 
-for VERIFY in $VERIFIES; do
-    VERIF=$(echo $VERIFY | tr '[:upper:]' '[:lower:]')
-    for MODE in $MODES; do
+for MODE in $MODES; do
+    PSK_TESTS=""
+    for VERIFY in $VERIFIES; do
+        VERIF=$(echo $VERIFY | tr '[:upper:]' '[:lower:]')
         for TYPE in $TYPES; do
+
+            if [ "$TYPE" = "PSK" ]; then
+                if [ -z "$PSK_TESTS" ]; then
```

Comment:
This code is a bit “clever”, so it requires at least a comment. For a given mode, we only run PSK tests for one value of VERIFY.

But can't we make the code simpler by iterating over MODE/TYPE/VERIFY rather than MODE/VERIFY/TYPE? Then, instead of iterating on VERIFY, we would iterate on an adjusted VERIFY which is normally VERIFY, but is `no` when `$TYPE = PSK`.

Reply:
makes sense. Done